fedimint-api/src/db/mod.rs

```diff
@@ -71,6 +96,64 @@ impl Database {
     pub async fn begin_transaction(&self) -> DatabaseTransaction {
         DatabaseTransaction::new(self.0.db.begin_transaction().await, &self.0.module_decoders)
     }
+
+    /// Runs a closure with a reference to a database transaction and tries to commit the
+    /// transaction if the closure returns `Ok` and rolls it back otherwise. If committing fails the
+    /// closure is run again for up to `retries` times. If retries is `None` it will run
+    /// `usize::MAX` times which is close enough to infinite times.
```

Comment:
running `usize::MAX` times was true at one point, but isn't anymore. We should probably make it true again since after `usize::MAX` rounds there will be an overflow/panic in debug mode and a silent overflow (afaik) in release mode.